### PR TITLE
[STN-62] Scaffold templates/patterns for HumSci Base

### PR DESCRIPTION
**Pull request by nrambeck**
# READY FOR REVIEW

# Summary

- Added the Drupal templates and UI Patterns from the legacy HumSci theme to support all current features.
- Also added configuration for Colorful theme block placement and configuration.
- re-organized the patterns folder to be a flat structure (no more atoms, molecules, organisms)
- Added more Drupal templates from Stanford Basic (blocks, menus, page title, html, node, page, region)
- Added favicon and screenshot (for Themes/Appearance UI)
- Removed Highlighted Region from theme (based on how the Stanford Basic theme is setup)
- Removed Google Analytics tag manager

# Steps to Test

If using Lando, prefix drush commands with `lando`.

1. Import configuration (`drush cim`) 
1. Clear caches (`drush cr`)
1. Enable the Colorful theme (`drush theme:enable humsci_colorful`)
1. Ensure various pages render without any errors
1. Compare Blocks placement on the Colorful theme to that of the legacy HumSci theme. Ensure they match.
1. Check `/patterns` page (that it works, but appears unstyled)

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr432-4do7iv3vnd05hj3cejqp9xj8mq4ckkrw.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0

Visual Diffs:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0/visualdiffs

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr432-4do7iv3vnd05hj3cejqp9xj8mq4ckkrw.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0

Visual Diffs:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0/visualdiffs

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr432-4do7iv3vnd05hj3cejqp9xj8mq4ckkrw.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0

Visual Diffs:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0/visualdiffs

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr432-4do7iv3vnd05hj3cejqp9xj8mq4ckkrw.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0

Visual Diffs:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0/visualdiffs

File: docroot/themes/humsci/humsci_basic/humsci_basic.info.yml [hunk 1]
```diff
@@ -15,3 +15,8 @@ regions:
   content: Content
   footer: Footer
   page_bottom: 'Page bottom'
+
+component-libraries:
+  humsci_basic:
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/humsci_basic.info.yml:19]: looks like .gitkeep files got added in this commit FYI

**Comment by bryanbraun** [comment]
To follow up on:

- Are there any generic templates from stanford_basic that we can include here as well?
- Can we bring over the favicon now?
- Theme preview image?
- Do we still need to use Atoms / Molecules / Organisms?

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr432-4do7iv3vnd05hj3cejqp9xj8mq4ckkrw.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0

Visual Diffs:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0/visualdiffs

**Comment by nrambeck** [comment]
I made the following changes based on feedback:
- re-organized the patterns folder to be a flat structure (no more atoms, molecules, organisms)
- Added more Drupal templates from Stanford Basic (blocks, menus, page title, html, node, page, region)
- Added favicon and screenshot (for Themes/Appearance UI)
- Removed Highlighted Region from theme (based on how the Stanford Basic theme is setup)

File: docroot/themes/humsci/humsci_basic/humsci_basic.info.yml (hunk 1 above)
Comment (nrambeck) [reply docroot/themes/humsci/humsci_basic/humsci_basic.info.yml:19]: I restructured this entire folder so we no longer having empty folders like this.

File: docroot/themes/humsci/humsci_basic/templates/block/block--system-menu-block--main.html.twig [hunk 2]
```diff
@@ -0,0 +1,37 @@
+{#
+/**
+ * @file
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/templates/block/block--system-menu-block--main.html.twig:3]: why do we have system-menu-block--main and system-menu-block below?

File: docroot/themes/humsci/humsci_basic/templates/contrib/patterns-variant-meta-information.html.twig [hunk 3]
```diff
@@ -0,0 +1,27 @@
+{#
+/**
+ * @file
+ * UI Pattern variant meta information.
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/templates/contrib/patterns-variant-meta-information.html.twig:4]: where did this file come from?

File: docroot/themes/humsci/humsci_basic/templates/menus/macros/nav-menu.twig [hunk 4]
```diff
@@ -0,0 +1,37 @@
+{#
+/**
+ * Macro for creating nested menus.
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/templates/menus/macros/nav-menu.twig:3]: where did this file come from as well? did you custom make it?

File: docroot/themes/humsci/humsci_basic/templates/page.html.twig [hunk 5]
```diff
@@ -1,45 +1,27 @@
-<header id="header">
```
Comment (silverli) [inline docroot/themes/humsci/humsci_basic/templates/page.html.twig]: so we switched this one out from the su_humsci theme one to the stanford_basic yes?

File: config/default/block.block.humsci_basic_messages.yml [hunk 6]
```diff
@@ -10,7 +10,7 @@ _core:
   default_config_hash: IN43gws2bI68lCkbtZPAxNJrQl7KqQJt2nhzkuJAqoM
 id: humsci_basic_messages
 theme: humsci_basic
-region: highlighted
```
Comment (silverli) [inline config/default/block.block.humsci_basic_messages.yml]: what is the reason for removing this?

**Comment by pookmish** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr432-4do7iv3vnd05hj3cejqp9xj8mq4ckkrw.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0

Visual Diffs:
- https://dashboard.tugboat.qa/5deead835590ff106b2b2ee0/visualdiffs

**Comment by nrambeck** [comment]
I added Masonry patterns back in and removed Google Tag Manage JS from the page template.

**Review by silverli — CHANGES_REQUESTED** [review]
I think this is just about ready, but I just have a couple further clarification questions I left in the comments that I'd like to go over Monday before merge.

File: config/default/block.block.humsci_basic_messages.yml (hunk 6 above)
Comment (silverli) [reply config/default/block.block.humsci_basic_messages.yml]: this region didn't exist in stanford basic, and it lived with help in su_humci

File: docroot/themes/humsci/humsci_basic/templates/menus/macros/nav-menu.twig (hunk 4 above)
Comment (silverli) [reply docroot/themes/humsci/humsci_basic/templates/menus/macros/nav-menu.twig:3]: it's from the latest stanford basic theme (4.x)

File: docroot/themes/humsci/humsci_basic/templates/contrib/patterns-variant-meta-information.html.twig (hunk 3 above)
Comment (silverli) [reply docroot/themes/humsci/humsci_basic/templates/contrib/patterns-variant-meta-information.html.twig:4]: this was in stanford basic. check for /patterns page when running locally

File: docroot/themes/humsci/humsci_basic/templates/page.html.twig (hunk 5 above)
Comment (nrambeck) [reply docroot/themes/humsci/humsci_basic/templates/page.html.twig]: Ending up using the latest Stanford Basic version of this template.

File: docroot/themes/humsci/humsci_basic/templates/block/block--system-menu-block--main.html.twig (hunk 2 above)
Comment (nrambeck) [reply docroot/themes/humsci/humsci_basic/templates/block/block--system-menu-block--main.html.twig:3]: The `--main` one represents the main content block only, which has no title in the block itself.

**Review by silverli — APPROVED** [review]
This looks great. Here are screenshots of the new theme working, just a bit unstyled on the home page and patterns page:

Screen Shot 2019-12-16 at 11 21 39 AM
Screen Shot 2019-12-16 at 11 22 03 AM

**Pull request merged**